meal.go

```diff
@@ -175,3 +178,144 @@ func getMealDataFromID(client HttpClient, id int) (week []MealData, err error) {
 
 	return
 }
+
+var regexDateFromTitle *regexp.Regexp = regexp.MustCompile("(\\d{1,2})\\D{1,2}(\\d{1,2})\\D{1,2}(\\d{1,2})\\D{1,2}(\\d{1,2})\\D?")
+
+// 현재 시간을 기준으로 해당 주의 시간표를 추출한다.
+func getMealDataWithTime(client HttpClient, now time.Time) (week []MealData, err error) {
+	defer WhereInError(&err, "현재 주간 학식")
+
+	weekday := now.Weekday()
+	if weekday < time.Monday || weekday > time.Friday {
+		err = NotFoundError.CreateError(errors.New("주말에는 학식이 제공하지 않습니다."))
+		return
+	}
+
+	// 학식 게시판 목록을 가져온다.
+	ids, err := getMealID(client)
+	if err != nil {
+		return
+	}
+
+	// 시간대 설정
+	loc, _ := time.LoadLocation("Asia/Seoul")
+
+	for _, id := range ids {
+		// 올린 날짜에서 년도를 가져오고
+		year, _ := strconv.Atoi(strings.Split(id.Date, "-")[0])
+
+		// 제목에서 기간을 가져온다.
+		parsedDuration := regexDateFromTitle.FindStringSubmatch(id.Title)
+		if len(parsedDuration) != 5 {
+			continue
+		}
+
+		// 가져온 기간을 숫자 배열로 다시 변환한다.
+		numberDuration := []int{}
+		for _, plain := range parsedDuration[1:] {
+			num, _ := strconv.Atoi(plain)
+			numberDuration = append(numberDuration, num)
+		}
+
+		// 년도와 기간을 통해 시작일과 종료일을 Time 자료형으로 생성한다.
+		startDay := time.Date(year, time.Month(numberDuration[0]), numberDuration[1], 0, 0, 0, 0, loc)
+		endDay := time.Date(year, time.Month(numberDuration[2]), numberDuration[3], 23, 59, 59, 99, loc)
+
+		// 현재 시간이 시작일과 종료일 사이에 있는지 확인한다.
+		if endDay.After(now) && startDay.Before(now) {
+			// 그 주의 학식을 가져온다.
+			week, err = getMealDataWithID(client, id.ID)
+			if err != nil {
+				return
+			}
+
+			return
+		}
+	}
+
+	err = NotFoundError.CreateError(errors.New("해당 기간에 학식이 존재하지 않습니다."))
```

Comment:
Codacy found an issue: [error strings should not be capitalized or end with punctuation or a newline](https://app.codacy.com/gh/s-owl/skhuinfo-api/pullRequest?prid=6091160)